packages/grafana-data/src/types/templateVars.ts

```diff
@@ -64,6 +65,16 @@ export interface AdHocVariableModel extends BaseVariableModel {
   baseFilters?: AdHocVariableFilter[];
 }
 
+export interface GroupByVariableModel extends BaseVariableModel {
```

Comment:
this is where i'm struggling most, what should this new model extend? i _think_ this variable model should extend `VariableWithMultiSupport` which extends `VariableWithOptions`, but then it's [forced to include a `query` key](https://github.com/grafana/grafana/blob/ash/group-by-poc-2/packages/grafana-data/src/types/templateVars.ts#L132) and also i'm not sure if it's possible for these multi value variables to have no value? 🤔

Reply:
@ashharrison90 I think extending `BaseVariableModel` is just fine. It's such a specific variable that I would keep the type dependency on the base type to a required minimum.

I think we also need to include `defaultOptions? : MetricFindValue[]` here to allow providing static configuration of the options.